ocaml/xapi-cli-server/records.ml

```diff
@@ -4976,3 +4979,37 @@ let repository_record rpc session_id repository =
           ()
       ]
   }
+
+let vtpm_record rpc session_id vtpm =
+  let _ref = ref vtpm in
+  let empty_record =
+    ToGet (fun () -> Client.VTPM.get_record rpc session_id !_ref)
+  in
+  let record = ref empty_record in
+  let x () = lzy_get record in
+  {
+    setref=
+      (fun r ->
+        _ref := r ;
+        record := empty_record
+      )
+  ; setrefrec=
+      (fun (a, b) ->
+        _ref := a ;
+        record := Got b
+      )
+  ; record= x
+  ; getref= (fun () -> !_ref)
+  ; fields=
+      [
+        make_field ~name:"uuid" ~get:(fun () -> (x ()).API.vTPM_uuid) ()
+      ; make_field ~name:"vm"
+          ~get:(fun () -> get_uuid_from_ref (x ()).API.vTPM_VM)
+          ()
+      ; make_field ~name:"profile"
```

Comment:
I think we need a way to change the profile as well. It is a bit like the `platform` field in the VM. Although I think we can only change it by resetting the vtpm, once the vtpm is in use I don't think that changing the profile is permitted. So maybe we need a way to specify the profile at `xe vtpm-create` time, is that possible?